File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        
        int x = 0;
        int y = 0;
        int i = 0;
        char c = ' ';
        while(i < s.length()){
            
            if(c == ' '){
                c = s.charAt(i);
            }
            
            if(c == s.charAt(i)){
                x++;
            }else{
                y++;
            }
            
            if(i == s.length() - 1){
                answer++;
                break;
            }
            
            if(x == y){
                answer++;
                x = 0;
                y = 0;
                c = ' ';
            }
            
            i++;
            
        }
        
        return answer;
    }
}

File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    case_1=0
    case_2=0
    for i in s:
        if case_1==case_2:
            answer+=1
            a=i
        if i==a:
            case_1+=1
        else:
            case_2+=1
    return answer